packages/pwa-kit-react-sdk/src/ssr/universal/components/with-react-query/index.js

```diff
@@ -13,16 +13,28 @@ import ssrPrepass from 'react-ssr-prepass'
 const isServerSide = typeof window === 'undefined'
 const STATE_KEY = '__reactQuery'
 
-export const withReactQuery = (Wrapped) => {
+export const withReactQuery = (Wrapped, options = {}) => {
     /* istanbul ignore next */
     const wrappedComponentName = Wrapped.displayName || Wrapped.name
+    const queryClientConfig = 
+        options.queryClientConfig || 
+        {
+            defaultOptions: {
+                queries: {
+                    retry: !isServerSide
+                },
+                mutations: {
+                    retry: !isServerSide
+                }
+            }
+        }
```

Comment:
FEEDBACK REQUIRED: 

Previously we had some discussions on whether or not we wanted retries to be disabled. I think we landed on the decision that we should have them disabled by default. At that point in time I thought that was in fact the default behaviour of react query hence why I thought it was a good idea. 

I may have backed out of my decision on wanted the default to be disabled in all cases to only have them disabled on server side. I chose this for the following reasons:

1. On the server we want to return a response as soon as possible, even if it is an error page.
2. On the client, retries maybe sense if you are going say from one page to the next, with known query parameters, e.g. a product id. If the query results in an error its probably not because a 404, but a network error. So it makes sense to retry.
3. For the cases that we don't want to retry maybe we explicitly have to set them on the hook level options?

Thoughts?

Reply:
**Context**
The `QueryClient` default global configuration is quite optimized for multiple scenarios.
We want to use the default global configuration baked in the library as much as we can and only deviate from the default values when we identify a scenario where is strictly necessary to make a change. And in that case, since it's a global configuration, we also need to think, verify and provide a solution for the rest of the scenarios affected by the change we made in the `QueryClient` default global configuration.

This new approach of using the Provider in the SDK to allow React Query to work server-side has some implications.

The Provider implemented in the SDK is global to all packages. That means the same Provider will be used by the `pwa-kit-react-sdk` package, the user space template aka `template-retail-react-app` package and the `commerce-sdk-react` package.

Decisions on the global configuration made by us inside the `pwa-kit-react-sdk` or made by in the user space inside their template will to affect all three packages.

**TL;DR**
My opinion is that we shouldn't change the default global configuration and adjust the configuration of the individual queries on a case basis using the configuration object.

We need to be aware of the potential consequences that global configuration changes will have in all three packages.
And we also need to think about the potential global configuration combinations a user can define in the user space, that will affect the server-side and client-side queries made by `commerce-sdk-react`.

Reply:
My vote is also to _not_ have our own default query client config. One of the reasons we chose to use react-query is its familiarity to many developers out there. I think we can assume that they would expect to see the defaults from react-query rather than from us.

Reply:
> we also need to think about the potential global configuration combinations a user can define in the user space, that will affect the server-side and client-side queries made by commerce-sdk-react.

🤔 that is possible eh? Since it looks like the QueryClient is being reused..

Reply:
@bendvc FYI there's also a separate (but related) discussion of when to do retries in this PR https://github.com/SalesforceCommerceCloud/pwa-kit/pull/718

Reply:
So my intentions here aren't to start maintaining a default configuration but solely to ensure retries don't occur on the server. 

With the default configurations behaviour requesting a page that has an error will mean that instead of rendering in 1 second it will render in 2+ seconds as it retries and fails 3 times. 

My opinion is that this is too much rope to give our developers, they will most definitely run into this scenario almost immediately.

Reply:
Please refer to [this](https://github.com/SalesforceCommerceCloud/pwa-kit/pull/734#discussion_r980578545) comment describing the decisions we made.